src/main/java/com/fasterxml/jackson/databind/ext/CoreXMLDeserializers.java

```diff
@@ -92,16 +92,50 @@ public Std(Class<?> raw, int kind) {
         public Object deserialize(JsonParser p, DeserializationContext ctxt)
             throws IOException
         {
-            // For most types, use super impl; but GregorianCalendar also allows
-            // integer value (timestamp), which needs separate handling
+            // GregorianCalendar also allows integer value (timestamp),
+            // which needs separate handling
             if (_kind == TYPE_G_CALENDAR) {
                 if (p.hasToken(JsonToken.VALUE_NUMBER_INT)) {
                     return _gregorianFromDate(ctxt, _parseDate(p, ctxt));
                 }
             }
+            // QName also allows object value, which needs separate handling
+            if (_kind == TYPE_QNAME) {
+                if (p.hasToken(JsonToken.START_OBJECT)) {
+                    return _parseQNameObject(p, ctxt);
+                }
+            }
             return super.deserialize(p, ctxt);
         }
 
+        private QName _parseQNameObject(JsonParser p, DeserializationContext ctxt)
+            throws IOException
+        {
+            JsonNode tree = ctxt.readTree(p);
+
+            if (!tree.has("localPart")) {
+                throw new JsonParseException("QName is missing required field: localPart");
```

Comment:
`JsonParseException` should be reserved for JSON decoding problems: here we need a `JsonMappingException` subtype. Usually via `DeserializationContext`, let me see what method should be used. Ideally `localPart` would also be included in quotes (since that's done with other messages)

Reply:
Ok, method to call is:

```
    public <T> T reportInputMismatch(Class<?> targetType,
            String msg, Object... msgArgs) throws JsonMappingException
```

I can maybe change it.

Reply:
Will change.